### PR TITLE
Add ES6 builds to libraries

### PR DESCRIPTION
**Pull request by eatyourgreens**
Add ES6 library builds with Babel:
- Enable the [React automatic runtime](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) for the classifier and Zooniverse React Components.
- Fix [`@babel/runtime`](https://babeljs.io/docs/en/babel-plugin-transform-runtime) in classifier builds.
- Add an ES6 build to `@zooniverse/react-components`. Export `dist/es6/index.js` as a module.
- Add an ES6 build to `@zooniverse/classifier`. Export `dist/es6/components/Classifier/index.js` as a module.
- Add `yarn bootstrap:es6` to bootstrap the monorepo with ES modules.
- Use `i18n.createInstance()` to set up `i18n` for each library. Give each instance its own `useTranslation` hook and `withTranslation` HOC.
- Use the ES6 build scripts for CI builds and for deploys. This might give us faster builds and smaller images.

Remove redundant packages from the builds:
- Replace `mime-types` with `mime/lite`.
- Remove `@fortawesome/react-fontawesome` and `@fortawesome/fontawesome-svg-core`.
- Remove `import * as d3`.
- Remove `import _ from lodash`.
- Replace `superagent` with `fetch` in the classifier.

I've left the Webpack [UMD module](https://jameshfisher.com/2020/10/04/what-are-umd-modules/) builds in for the Mocha tests, and for backwards compatibility with any third-party projects which `require` in `@zooniverse/react-components`. If we can figure out #3779 then I'm fairly sure we can remove the webpack builds and use pure ES modules for the libraries.

With the changes here, I'm seeing the JS download on a typical project Classify page drop from ~1.7MB to ~800k. NextJS reports the project page build size as ~490k (Classify pages are ~800k), which is still too large but much better than what we've currently got in production.

- Includes #3847.
- Towards #366.

Lots of files are touched by this PR, but many of them are one-line changes to the imports from `i18next` packages.

## Package
app-project
lib-classifier
lib-react-components

## How to Review
There are a few options. Pick the one you're most comfortable with:
- [Build and start both NextJS apps locally in Node](https://github.com/zooniverse/front-end-monorepo/#with-node-and-yarn) to verify that they still build and run. You can run `yarn bootstrap:es6` to skip the Webpack builds, for a faster build that only includes the changes here.
- [Build and start a Docker image](https://github.com/zooniverse/front-end-monorepo/#docker), then check both NextJS apps on localhost.
- [Build and deploy a Docker image of this branch](https://github.com/zooniverse/front-end-monorepo/actions/workflows/deploy_branch.yml), then check the project app on https://fe-project-branch.preview.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess. There's no branch preview on kubernetes for the content pages app.

I've run a branch deploy of 504229f49b22b5e82861c57872ac46fb993a773e so that we can test the Docker image on kubernetes.

Here are some test projects:
- SuperWASP Black Hole Hunters: https://fe-project-branch.preview.zooniverse.org/projects/hughdickinson/superwasp-black-hole-hunters/classify/workflow/17386
- Planet Hunters TESS in Spanish: https://fe-project-branch.preview.zooniverse.org/projects/es/nora-dot-eisner/planet-hunters-tess
- HMS NHS: https://fe-project-branch.preview.zooniverse.org/projects/msalmon/hms-nhs-the-nautical-health-service
- Staging workflows with feedback enabled: https://fe-project-branch.preview.zooniverse.org/projects/markb-panoptes/feedback-testing-mb?env=staging
- You can verify the deployed commit in the `<head>` of each page, or at the following URL: https://fe-project-branch.preview.zooniverse.org/projects/commit_id.txt

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [x] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [x] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out
- [x] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [x] The PR creator has described the reason for refactoring
- [x] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Tests are passing and the apps are building, so that looks good.

Dynamic imports seem to be broken in the app bundles. In the project app:
- dynamically imported translation text is missing from the page header and footer.
- the `Classifier` component isn't dynamically loaded into the Classify page.

The content pages app looks good, but it doesn't use internationalisation.

**Comment by eatyourgreens** [comment]
Tests are passing and both NextJS apps are building. I've deployed the project app to kubernetes. It loads and runs without errors, but translated labels don't show for components that are loaded in from `@zooniverse/react-components`.
https://fe-project-branch.preview.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess/classify/workflow/11235

**Comment by eatyourgreens** [comment]
The project storybook doesn't show translation strings on this branch, which is odd because this PR doesn't touch project translations.

The story for project recent classifications, showing translation keys instead of text labels.

**Comment by eatyourgreens** [comment]
This is working locally for me now without translations bugs, in production mode. The storybooks are working too. I've run a branch deploy for testing of the Docker image on kubernetes.

**Comment by eatyourgreens** [comment]
Still to do: update the changelog for `@zooniverse/react-components`.

**Comment by eatyourgreens** [comment]
Also to do: roll #3847 into this branch, to fix `@babel/runtime` in the classifier.

**Review by mcbouslog — APPROVED** [review]
This looks great!!

I tested all the linked `fe-project-branch` projects, noting changing languages, signing in and out, feedback (positive and negative), d3 related subject viewers - all work as expected. Tested across browsers without issue.

I successfully built and ran app-content-pages and app-project locally. Also ran the storybooks locally; spot-checked a few stories including test translation without issue.

I can sign in/out on the `fe-project-branch` linked projects.

Looks great!!

File: packages/lib-react-components/src/translations/i18n.js [hunk 1]
```diff
@@ -26,7 +26,8 @@ const supportedLngs = [
   'zh-tw'
 ]
 
-i18n.use(initReactI18next).init({
+const zrcI18n = i18n.createInstance()
```
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/lib-react-components/src/translations/i18n.js:29]: This library is the replacement for the old ZRC library, in new projects, so I think the naming is OK. 

Something I discovered yesterday is that the build I added here (straight Babel transform from JSX to JS) is the build script from the old library. I’m not sure why it’s taken so long to copy this across to the new libraries. 

https://github.com/zooniverse/Zooniverse-React-Components/blob/49bea57c6089bb99e51f2552b148728f96960705/package.json#L9

[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/lib-react-components/src/translations/i18n.js:29]: Still kind of confusing that there are two ZRC libraries in NPM. 
- https://www.npmjs.com/package/zooniverse-react-components
- https://www.npmjs.com/package/@zooniverse/react-components

**Pull request merged**